### PR TITLE
data registry user permissions role changes

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
Addition of permissions for data registries. This PR only includes the role UI and data model changes.

Peek 2021-09-08 15-27

There are two new permissions, 'manage registries' and 'view registry contents'. Both of these can be set on a per-registry basis. 

*Manage Registries*: The intention of this permission is to give users access to all the management functions of data registries including invitations, grants etc.

*View Registry Contents*: This permission gives users access to the data 'contained' in the registry.

I went with a slightly different UI for these permissions which I think makes it easier to understand the selections the user is making. I'm interested in feedback on this UI.

## Feature Flag
DATA_REGISTRY

## Product Description
Permissions for managing data registries and access to data registry data.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
None

### QA Plan
Ticket to follow

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Comment by orangejenny** [comment]
I like the none/some/all dropdown. The language & flow are more natural than the current UI where you uncheck an "all" checkbox and then the list of reports appears.

FYI @biyeun

File: corehq/apps/users/static/users/js/roles.js [hunk 1]
```diff
@@ -335,12 +335,33 @@ hqDefine('users/js/roles',[
                     data.name = data.name.trim();
                 }
 
-                data.permissions.view_report_list = ko.utils.arrayMap(ko.utils.arrayFilter(data.reportPermissions.specific, function (report) {
-                    return report.value;
-                }), function (report) {
-                    return report.path;
-                });
-                data.permissions.view_reports = data.reportPermissions.all;
+                const unWrapParameterizedPermission = function (permission, item_attr='slug') {
```
Comment (orangejenny) [inline corehq/apps/users/static/users/js/roles.js]: nitpick: `unWrap` => `unwrap`

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/users/static/users/js/roles.js [hunk 2]
```diff
@@ -254,18 +361,26 @@ hqDefine('users/js/roles',[
                     data.name = data.name.trim();
                 }
 
-                data.permissions.view_report_list = ko.utils.arrayMap(ko.utils.arrayFilter(data.reportPermissions.specific, function (report) {
-                    return report.value;
-                }), function (report) {
-                    return report.path;
-                });
+                const unWrapItemList = function (items, item_attr = 'slug') {
```
Comment (snopoke) [inline corehq/apps/users/static/users/js/roles.js:364]:
```suggestion
                const unwrapItemList = function (items, item_attr = 'slug') {
```

File: corehq/apps/users/static/users/js/roles.js (hunk 1 above)
Comment (snopoke) [reply corehq/apps/users/static/users/js/roles.js]: I thought that looked funny but couldn't put my finger on it. Thanks: a1f5de2014b

**Pull request merged**